Clockify.Tests/Tests/ReportTests.cs

```diff
@@ -0,0 +1,118 @@
+            {
+                Name = "GetDetailedReportAsync " + guid,
+                Color = "#FF00FF",
+                HourlyRate = new Net.Models.HourlyRates.HourlyRateRequest { Amount = hourlyRateAmount },
+                ClientId = createClientResponse.Data.Id
+            };
+
+            var createProject = await _client.CreateProjectAsync(_workspaceId, projectRequest);
+            createProject.IsSuccessful.Should().BeTrue();
+            createProject.Data.Should().NotBeNull();
+
+            ProjectDtoImpl project = createProject.Data;
+
+            var timeEntryRequest = new TimeEntryRequest
+            {
+                Start = now,
+                End = now.AddSeconds(30),
+                ProjectId = project.Id
+            };
+
+            var createResult = await _client.CreateTimeEntryAsync(_workspaceId, timeEntryRequest);
+            createResult.IsSuccessful.Should().BeTrue();
+
+            var userResponse = await _client.GetCurrentUserAsync();
+            userResponse.IsSuccessful.Should().BeTrue();
+
+            // First, obtain the OS version of time zone based on the Clockify users' settings.
+            var tzi = TZConvert.GetTimeZoneInfo(userResponse.Data.Settings.TimeZone);
+
+            // Second, translate current time into the Clockify users' time zone.
+            var nowTz = now.ToOffset(tzi.BaseUtcOffset).DateTime;
+
+            // Third, just to be safe we need to translate again to make sure Daylight Savings time is accounted for.
+            var nowTz2 = now.ToOffset(tzi.GetUtcOffset(nowTz)).DateTime;
+
+            var detailedReportRequest = new DetailedReportRequest
+            {
+                ExportType = ExportType.JSON,
+                DateRangeStart = nowTz2.AddMinutes(-2),
+                DateRangeEnd = nowTz2.AddMinutes(2),
+                SortOrder = SortOrderType.DESCENDING,
+                Description = String.Empty,
+                Rounding = false,
+                WithoutDescription = false,
+                AmountShown = AmountShownType.EARNED,
+                Clients = new ClientsFilterDto
+                {
+                    Contains = ContainsType.CONTAINS,
+                    Ids = new System.Collections.Generic.List<string> { client.Id },
+                    Status = StatusType.ACTIVE
+                },
+                DetailedFilter = new DetailedFilterDto
+                {
+                    SortColumn = SortColumnType.DATE,
+                    Page = 1,
+                    PageSize = 50
+                },
+                TimeZone = userResponse.Data.Settings.TimeZone
+            };
+
+            var getDetailedReportResult = await _client.GetDetailedReportAsync(_workspaceId, detailedReportRequest);
+            getDetailedReportResult.IsSuccessful.Should().BeTrue();
+            getDetailedReportResult.Data.Should().NotBeNull();
+            getDetailedReportResult.Data.Timeentries.Count.Should().BeGreaterOrEqualTo(1);
```

Comment:
I think that FluentAssertion have method with syntax `getDetailedReportResult.Data.Timeentries.HaveCountSomethingSomething()`. That would be better if possible. If not let's leave it.

Reply:
`getDetailedReportResult.Data.Timeentries.Should().HaveCountGreaterOrEqualTo(1)` -< That one should work. Let me know if you can see it.

Reply:
Yes, I can see that now. Not sure why I couldn't find it yesterday. Thanks for pointing it out!